### ISSUE TITLE
Absolute paths with hyphens don't work in function templates

### ISSUE THREAD

**Issue by neilpa** [RhetTbull/osxphotos#477]
Ran into an interesting bug today when trying to make my export script runnable from any directory. Looks like there's an issue with the template parser that causes it to choke on file path strings that include a hyphen.

```sh
osxphotos export \
  --directory '{function:/Volumes/sandisk-hfsp/bin/pics_templates.py::export_directory}' \
  --filename '{function:/Volumes/sandisk-hfsp/bin/pics_templates.py::export_filename}' \
...
Error: Invalid template '{function:/Volumes/sandisk-hfsp/bin/pics_templates.py::export_filename}': SyntaxError: None:1:27: error: Expected SUBFIELD_WORD or '|' or '(' or '[' or ' ' or '?' or ',' or '}' at position (1, 27) => 'es/sandisk*-hfsp/bin/'.
```

I was able to work around this for now by moving my template files. Hoping this can be fixed though since I run everything off an external SSD and the volume name includes a hyphen.

**Comment by RhetTbull** [comment]
I think this is fixed in [v0.42.46](//github.com/RhetTbull/osxphotos/releases/tag/v0.42.46).  It would have failed for a lot of other valid path characters too.  Due to the template syntax, template function paths cannot include `,|[]()` but other characters should be OK.

**Comment by neilpa** [comment]
Awesome!

> template function paths cannot include `,|[]()`

If you've got any of those characters in filepaths you have bigger issues...
